test/test_transforms_v2_refactored.py

```diff
@@ -2280,3 +2280,34 @@ def resize_my_datapoint():
         _register_kernel_internal(F.resize, MyDatapoint, datapoint_wrapper=False)(resize_my_datapoint)
 
         assert _get_kernel(F.resize, MyDatapoint) is resize_my_datapoint
+
+
+def test_operations():
+
+    img = datapoints.Image(torch.rand(3, 10, 10))
+    t = torch.rand(3, 10, 10)
+    mask = datapoints.Mask(torch.rand(1, 10, 10))
+
+    for out in [img + t, t + img, img * t, t * img, img + 3, 3 + img, img * 3, 3 * img, img + img]:
+        assert isinstance(out, datapoints.Image)
+
+    for out in [mask + t, t + mask, mask * t, t * mask, mask + 3, 3 + mask, mask * 3, 3 * mask, mask + mask]:
+        assert isinstance(out, datapoints.Mask)
+
+    with pytest.raises(TypeError, match="unsupported operand type"):
+        img + mask
+
+    with pytest.raises(TypeError, match="unsupported operand type"):
+        img * mask
+
+    bboxes = datapoints.BoundingBoxes(
+        [[17, 16, 344, 495], [0, 10, 0, 10]],
+        format=datapoints.BoundingBoxFormat.XYXY,
+        canvas_size=(1000, 1000)
+    )
+    t = torch.rand(2, 4)
+
+    for out in [bboxes + t, t + bboxes, bboxes * t, t * bboxes, bboxes + 3, 3 + bboxes, bboxes * 3, 3 * bboxes, bboxes + bboxes]:
+        assert isinstance(out, datapoints.BoundingBoxes)
+        assert not(hasattr(out, "format"))
+        assert not(hasattr(out, "canvas_size"))
```

Comment:
We're losing the meta-data on the BBox instance. That's OK: if the meta-data is needed later, there will be a loud and clear error message.
Another way to look at it is that in `main` we return a Tensor, which still misses the meta-data, and still leads to an error in the exact same cases.